### ISSUE TITLE
Samplesheets: Re-initialize Vue app and update breaking dependencies

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#837]
Tl;dr: Noticed that simply trying to upgrade the existing setup's dependencies didn't work, or at least would require a whole lot of work. The old setup used an old version of vue-cli and Webpack, the latter of which would be a real pain to upgrade manually.

I decided to re-start the vue app with the new vue-cli, which abstracts away Webpack configuration and is just nicer in general. Also easier to integrate with Django.

While doing this, I'm taking this chance to move the app to `samplesheets/vueapp`, as currently the Django and Vue files are mixed up and all messy. Also, I've reintroduced unit tests with jest and should configure some tests soon.

## Issues/TODO

- [x] Set up production build
- [x] Some CSS stuff isn't working correctly (1)
- [x] Pycharm is now linting Django templates with eslint
- [x] A ton of new eslint warnings need to be fixed (related to #537)
- [x] UI tests failing randomly

## Notes

- (1) See e.g. iRODS buttons, nav bar buttons and popup unit editor.
    * This is weird as I'm using the same version of ag-grid and bootstrap-vue as before the upgrade..
    * **Update:** Turns out of these issues are not CSS related at all! Looks like whitespace rendering for Vue templates has changed in the setup I'm currently using. Previously whitespace characters entered "margins" where there actually are none..
- For `webpack-bundle-tracker` the default version currently installed is `v1.1.0-alpha`. That doesn't work with `django-webpack-loader` due to [this issue](https://github.com/owais/django-webpack-loader/issues/227)
    * Installing `v0.4.3` should be ok until a fix is merged into the sable django loader.
- The new Vue CLI has abstracted away Webpack configuration. Webpack settings should now go into `vue.config.js`, [see here for an example](https://cli.vuejs.org/guide/webpack.html#simple-configuration)
- [Django & Webpack instructions](https://gist.github.com/lsapan/bdc1fbba71058cc337abc181a84b4f49) updated for vue-cli v4!
